### ISSUE TITLE
magic-number-prevention scans inside string literals and does not accept an object property as a name (217 hits here, 0 real)

### ISSUE THREAD

**Issue by ruleofcodedev** [Rule-Of-Code/RoC#105]
**Version:** 7.21.0
**Law:** `magic-number-prevention`

Not a duplicate of #68 (HH:MM:SS exemption) or #69 (`lawId` not threaded, which
is fixed — `ignores.byRule` works for this law now). This is about where the
detector looks.

## What happens

12 files reported. Across all of them, **217 numbers are flagged and none is a
magic number**:

```
flagged as the law counts them : 217
  inside a string literal      : 153
  a named object property      :  64
  neither — a real candidate   :   0
```

Measured by re-running the law's own detector and classifying each hit by
position. Script at the end.

## Why

```js
// checkers/code-quality-laws/magic-number-prevention.js:87
const code = this.stripComments(content);
const numberRegex = /\b\d+\.\d+\b|\b\d{2,}\b/g;
```

Comments are stripped. **String literals are not**, so every digit inside quoted
text is a candidate:

```ts
{ id: 'cfg', n: '01', title: 'The config file' }        // docs.data.ts — '01'..'09'
{ t: `"✅ alert policy: unauthorized_401"` }             // findings.field.ts — 401
`laws.minLawsChecked: 999` // inside a captured CLI transcript              // start.ts
```

The only "this number is named" exemption is a variable declaration:

```js
/\b(?:const|let|var|readonly)\s+[A-Za-z_$][\w$]*\s*=\s*-?$/
```

An object property is not one, so a value whose name sits immediately to its left
is still reported:

```ts
const HERO = { curves: 44, ratio: 7, innerCurves: 26, radiusFactor: 0.62 };  // guilloche.ts
{ platform: 'Angular', laws: 31 }                                            // adopt.platform-fe.ts
{ consumer: '3Trade', lawsEnforced: 93, reportDate: 'July 2026' }            // in-production.ts
```

`curves: 44` is exactly what the law asks for — a number with a name attached.
The comment above that object in our source reads "Every ratio is named: the
engraving is maths, not magic", written before we knew the law could not see it.

## Suggested direction

Two changes, either of which removes most of the noise:

1. **Blank string and template literals before scanning**, the way comments
   already are. A number inside quoted text is data or prose; it can never be
   replaced by a named constant, so it can never be actioned.
2. **Accept an object property as a name.** Extend the left-context test:

   ```js
   /(?:\b(?:const|let|var|readonly)\s+[A-Za-z_$][\w$]*|[A-Za-z_$][\w$]*)\s*[:=]\s*-?$/
   ```

   That covers `curves: 44`, `laws: 31` and `x = 44` with one pattern.

(1) alone drops 153 of our 217; both together drop all of them, and in this
repository that is the right answer — there is no unexplained constant in the
files reported.

## Script

```js
// classify each hit the law makes, by position
const NUMBER = /\b\d+\.\d+\b|\b\d{2,}\b/g;
const ACCEPTABLE = new Set(['0','1','-1','2','10','100','1000']);
const NAMED_CONST = /\b(?:const|let|var|readonly)\s+[A-Za-z_$][\w$]*\s*=\s*-?$/;
const OBJECT_PROPERTY = /[A-Za-z_$][\w$]*\s*:\s*-?$/;
// blankStrings() replaces every string/template literal with spaces of equal
// length, so a hit that disappears was inside one.
```

**Comment by ruleofcodedev** [comment]
Fixed in #112, merged to `develop`. Ships in the next release.

Both of your suggested changes went in, because each removes a different half of the noise. Measured on your tree:

```
12 files, 217 numbers  ->  0
```

Your classification was exact — 153 inside string literals, 64 named object properties, none real. And the comment above your `HERO` object, *"Every ratio is named: the engraving is maths, not magic"*, was the argument: `{ curves: 44 }` is the fix this law asks for, so reporting it left nothing to do.

Two existing tests of ours asserted the opposite — `{ limit: 500 }` and `{ retries: 5000 }` should be reported. They were changed deliberately rather than repaired, and that is called out as a behaviour change in the changelog.

**One thing worth reporting back, because it was our defect and it was worse than the one you filed.** The string-blanking I added read `/['"]/` — a regex containing a quote — as the start of a string, and blanked everything up to the next quote in the file. That turns a false positive into a false **negative**, invisibly. Fixed in #113 before either shipped: regex literals are skipped whole, division is told from a regex by context, and character classes and escaped slashes are honoured.

I also checked the 41 findings that disappeared from our own repository, file by file against their real source lines, to be sure the widening had not hidden anything real. All of them are inside strings or comments.
